src/bin/ion/commands/beta/generate/mod.rs

```diff
@@ -102,7 +101,14 @@ impl IonCliCommand for GenerateCommand {
         println!("Started generating code...");
 
         // generate code based on schema and programming language
-        CodeGenerator::new(language, output).generate(schema)?;
+        match language {
+            "java" => CodeGenerator::<JavaLanguage>::new(output).generate(schema)?,
+            "rust" => CodeGenerator::<RustLanguage>::new(output).generate(schema)?,
+            _ => bail!(
+                "Unsupported programming language: {}, this tool only supports Java and Rust code generation.",
```

Comment:
```suggestion
                "Programming language '{}' is not yet supported. Currently supported targets: 'java', 'rust'",
```

This error message:
* Suggests that the requested programming language could be added in the future
* Tells the user what strings the CLI does recognize. (In case the user put in something conceptually correct like `.rs`, `rustlang` or `jvm`.)